core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -2285,6 +2290,18 @@ private CompiledQuery repairTables(SqlExecutionContext executionContext) throws
         return compiledQuery.ofRepair();
     }
 
+    private void schedulePurgeO3Partitions(CharSequence tableName, int partitionBy) {
+        final MPSequence seq = messageBus.getO3PurgeDiscoveryPubSeq();
+        long cursor = seq.next();
+        if (cursor > -1) {
+            O3PurgeDiscoveryTask task = this.messageBus.getO3PurgeDiscoveryQueue().get(cursor);
+            task.of(tableName, partitionBy);
+            seq.done(cursor);
+            return;
+        }
+        throw CairoException.instance(0).put("cannot schedule vacuum action, queue is full");
```

Comment:
can this be SQLException to make it consistent with invocation site?

Reply:
It used to SqlException but then I changed to CairoException deliberately. Sql means syntax, query position etc, nothing like that in here

Reply:
There is nothing wrong with query, position is irrelevant. Action retry is also not something SQL exception should tell. 
This method on the surface can throw only SQL exception but opening reader can easily throw `CairoException`, this is same to me as  not being able to open the reader. Wait, retry, change configuration, restart server etc

Reply:
ok, I changed https://github.com/questdb/questdb/pull/1778/commits/3c7038f654b3e9cadc28e6d99accf84451c7c729